File: Code Jam 2022/Qualification Round/d1000000/d1000000.py
#!/usr/bin/env python3


if __name__ == "__main__":
    for test_case in range(1, int(input())+1):
        n = int(input())
        nums = sorted(map(int, input().split()), reverse=True)

        for idx, value in enumerate(nums):
            if value + idx < n:
                n = value + idx
                nums = nums[:n]

        print(f'Case #{test_case}: {len(nums)}')

File: Code Jam 2022/Qualification Round/d1000000/d1000000.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var testCases int
	fmt.Scanln(&testCases)

	for testCase := 1; testCase <= testCases; testCase++ {
		var n int
        fmt.Scanln(&n)
        
        numbers := make([]int, n)
        for i := range numbers {
            fmt.Scan(&numbers[i])
        }
        
        // numbers := []int{10, 10, 7, 6, 7, 4, 4, 5, 7, 4}
        
        sort.Sort(sort.Reverse(sort.IntSlice(numbers)))
        
        for idx, value := range numbers {
            if value < n - idx {
                n = value + idx
                numbers = numbers[:n]
            }
        }
        
        fmt.Printf("Case #%d: %d\n", testCase, len(numbers))
	}
}
